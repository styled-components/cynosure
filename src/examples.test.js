#!/bin/env node

const path = require('path')
const getExamples = require('../helpers').getExamples

getExamples('./').forEach((exampleFile) => {
  describe(`${exampleFile.substring(0, exampleFile.indexOf('.examples.js'))} examples`, () => {
    // eslint-disable-next-line global-require, import/no-dynamic-require
    const examples = require(path.resolve('./', exampleFile))
    examples.forEach((example, index) => {
      it(`should match example #${index}`, () => {
        expect(example).toMatchSnapshot()
      })
    })
  })
})
