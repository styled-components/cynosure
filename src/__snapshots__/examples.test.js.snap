exports[`lib/helpers/examples/stripUnit examples should match example #0 1`] = `
Object {
  "--dimension": 100,
}
`;

exports[`lib/helpers/examples/stripUnit examples should match example #1 1`] = `
Object {
  "--dimension": 100,
}
`;

exports[`src/helpers/examples/stripUnit examples should match example #0 1`] = `
Object {
  "--dimension": 100,
}
`;

exports[`src/helpers/examples/stripUnit examples should match example #1 1`] = `
"
  --dimension: 100
"
`;
